[Bug] vela top crash, when view Application topology

**Describe the bug**
<!--
A clear and concise description of what the bug is.
-->
`vela top` crash, when view Application's Topology.





**To Reproduce**
<!--
Steps to reproduce the behavior:
1. The YAML files of Component/Trait I used.
2. The YAML file of Application I applied.
3. Other operations I have done.
4. See error
-->

1. apply one Application with customized Trait
2. `vela top` to show this Application, its status is rending, because vale can not find the Customized Trait
```
status:
  conditions:
  - lastTransitionTime: "2023-06-15T10:37:42Z"
    message: 'component(xxxxxxxxx) parse trait(xxxxxxx):
      trait definition of xxxxx not found'
    reason: ReconcileError
    status: "False"
    type: Parsed
  observedGeneration: 1
  status: rendering
```
4. press `t` to view Application topology,  vela crashs.
```
[root@I ~]# vela top
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x70 pc=0x29d9e4a]

goroutine 1 [running]:
github.com/rivo/tview.(*Application).Run.func1()
        /home/runner/go/pkg/mod/github.com/rivo/tview@v0.0.0-20221128165837-db36428c92d9/application.go:244 +0x4d
panic({0x2e8ab40, 0x5732ad0})
        /opt/hostedtoolcache/go/1.19.9/x64/src/runtime/panic.go:884 +0x212
github.com/oam-dev/kubevela/references/cli/top/view.(*TopologyView).NewAppTopologyView(0xc000a86f50)
        /home/runner/work/kubevela/kubevela/references/cli/top/view/application_topology_view.go:297 +0x8ca
github.com/oam-dev/kubevela/references/cli/top/view.(*TopologyView).Update.func1()
        /home/runner/work/kubevela/kubevela/references/cli/top/view/application_topology_view.go:140 +0x68
github.com/oam-dev/kubevela/references/cli/top/view.(*TopologyView).Update(0xc000a86f50, 0x365d450)
        /home/runner/work/kubevela/kubevela/references/cli/top/view/application_topology_view.go:149 +0x250
github.com/oam-dev/kubevela/references/cli/top/view.(*TopologyView).Start(0xc000a86f50)
        /home/runner/work/kubevela/kubevela/references/cli/top/view/application_topology_view.go:110 +0x25
github.com/rivo/tview.(*Application).QueueUpdateDraw.func1()
        /home/runner/go/pkg/mod/github.com/rivo/tview@v0.0.0-20221128165837-db36428c92d9/application.go:755 +0x26
github.com/rivo/tview.(*Application).Run(0xc000518460)
        /home/runner/go/pkg/mod/github.com/rivo/tview@v0.0.0-20221128165837-db36428c92d9/application.go:381 +0xa42
github.com/oam-dev/kubevela/references/cli/top/view.(*App).Run(0xc000d0a9b0)
        /home/runner/work/kubevela/kubevela/references/cli/top/view/app.go:115 +0x77
github.com/oam-dev/kubevela/references/cli.launchUI({0xc0010cb200, 0x0, 0xc000376770, {0x3b40550, 0xc00017bf10}, {0x0, 0x0}, 0x0, 0x0}, {0x340b676, ...})
        /home/runner/work/kubevela/kubevela/references/cli/top.go:81 +0x105
github.com/oam-dev/kubevela/references/cli.NewTopCommand.func1(0xc0001c0900?, {0x3403288?, 0x0?, 0x0?})
        /home/runner/work/kubevela/kubevela/references/cli/top.go:57 +0x165
github.com/spf13/cobra.(*Command).execute(0xc0001c0900, {0x57d1618, 0x0, 0x0})
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:916 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0xc000be6000)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
main.main()
        /home/runner/work/kubevela/kubevela/references/cmd/cli/main.go:41 +0x18c
```

**Expected behavior**
<!--
A clear and concise description of what you expected to happen.
-->

**Screenshots**
<!--
If applicable, add screenshots to help explain your problem.
-->

**KubeVela Version**

<!--
Describe your KubeVela controller or CLI version information.
-->

**Cluster information**
<!--
Describe your kubernetes cluster information.
- Kubernetes Version [e.g. 1.16.9]
-->

**Additional context**
<!--
Add any other context about the problem here.
-->

--- a/references/cli/top/view/application_topology_view.go
+++ b/references/cli/top/view/application_topology_view.go
@@ -292,15 +292,17 @@ func (v *TopologyView) NewAppTopologyView() *TopologyTree {
 		return newTopology
 	}
 	// workflow
-	workflowNode := tview.NewTreeNode(v.formatter.EmojiFormat("WorkFlow", "workflow")).SetSelectable(true)
-	root.AddChild(workflowNode)
-	for _, step := range app.Status.Workflow.Steps {
-		stepNode := tview.NewTreeNode(component.WorkflowStepFormat(step.Name, step.Phase))
-		for _, subStep := range step.SubStepsStatus {
-			subStepNode := tview.NewTreeNode(subStep.Name)
-			stepNode.AddChild(subStepNode)
+	if app.Status.Workflow != nil {
+		workflowNode := tview.NewTreeNode(v.formatter.EmojiFormat("WorkFlow", "workflow")).SetSelectable(true)
+		root.AddChild(workflowNode)
+		for _, step := range app.Status.Workflow.Steps {
+			stepNode := tview.NewTreeNode(component.WorkflowStepFormat(step.Name, step.Phase))
+			for _, subStep := range step.SubStepsStatus {
+				subStepNode := tview.NewTreeNode(subStep.Name)
+				stepNode.AddChild(subStepNode)
+			}
+			workflowNode.AddChild(stepNode)
 		}
-		workflowNode.AddChild(stepNode)
 	}
 
 	// component

